### PR TITLE
Implement ActionInterface for /search/term/popular

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This is a part of the process of introducing implementations of `ActionInterface`.

### Fixed Issues (if relevant)

N/A

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            ->with(static::XML_PATH_SEO_SEARCH_TERMS, ScopeInterface::SCOPE_STORE)
+            ->willReturn(true);
+        $resultPageMock = $this->getMockBuilder(ResultPage::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->resultPageFactoryMock->expects($this->once())
+            ->method('create')
+            ->willReturn($resultPageMock);
+
+        $this->action->execute();
```
Comment (ihor-sviziev) [inline app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php]: I think it’s good idea to check that result is result page mock, just to make sure we got correct result

File: app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->resultForwardFactoryMock->expects($this->once())
+            ->method('create')
+            ->willReturn($resultForwardMock);
+        $resultForwardMock->expects($this->once())
+            ->method('forward')
+            ->with('noroute');
+
+        $this->action->execute();
```
Comment (ihor-sviziev) [inline app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php]: I think it’s good idea to check that result is result forward mock, just to make sure we got correct result

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php (hunk 1 above)
Comment (Bartlomiejsz) [reply app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php]: Good point, done!

File: app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php (hunk 2 above)
Comment (Bartlomiejsz) [reply app/code/Magento/Search/Test/Unit/Controller/Term/PopularTest.php]: Good point, done!

**Comment by ihor-sviziev** [comment]
@Bartlomiejsz usually for objects using `assersSame` method. Did you checked that your test works correct?

**Comment by Bartlomiejsz** [comment]
@ihor-sviziev yes, it works for both. I modified it to `assertSame` :)

**Comment by ihor-sviziev** [comment]
Looks good. I'll review this PR after all tests will be finished

**Review by ihor-sviziev — APPROVED** [review]
We have quite a lot functional tests failures, but seems like they are not related to changes from this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7220](https://jira.corp.magento.com/browse/ENGCOM-7220) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
Failed functional tests not related to the changes in this PR

**Pull request merged**